# Tehtävä 11 peliteoria

Lähtötiedot
Seuraavassa ovat yritysten A ja B keskinäisen kilpailun strategioiden kuvaukset.

Strategiat ovat yrityksen A näkökulmasta alla olevat. Luvut ovat markkinaosuuksien muutoksia viikoittain. Kyse on nollasummapelistä, jossa B:n hyödyt ovat käänteisiä A:n arvoille.

Strategiat yrityksen A näkökulmasta


| **Yrityksen A strategiat** | **C1** | **C2** | **C3** |
|----------------------------|--------|--------|--------|
| **S1**                     | 3      | -2     | -1     |
| **S2**                     | -1     | 0      | 2      |
| **S3**                     | -1     | 1      | -1     |
| **S4**                     | -1     | 2      | -3     |

**Yrityksen B strategiat** ovat C1, C ja C3.


Kuvaa oheinen tilanne NashPy-kirjaston pelinä (A, B).

Kuvaa pelin tuloksia joillain valitsemillasi A:n ja B:n strategioilla σA ja σB.

Laske pelin Nashin tasapaino. Mitä laskemasi luvut merkitsevät?

Piirrä myös kuvioon kuvitteellisen pelin mukainen tilanne tai tilanteita, joissa kuvataan toistojen vaikutusta todennäköisyyksien suppenemiseen tai hajaantumiseen, eli siihen, lähestyvätkö ne jotain yhtä tiettyä arvoa.

In [1]:
# Kirjastojen tuonti

import nashpy as nash
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# strategiamatriisit
payoff_A = np.array([[3, -2, -1],
                     [-1, 0, 2],
                     [-1, 1, -1],
                     [-1, 2, -3]])

payoff_B = -payoff_A  # nollasumma

In [3]:
# pPelin objekti NashPy-kirjastossa
game = nash.Game(payoff_A, payoff_B)

In [4]:
# Nashin tasapaino
equilibria = list(game.support_enumeration())

In [5]:
print("Lasketut Nashin tasapainot:")
for eq in equilibria:
    print(eq)

Lasketut Nashin tasapainot:
(array([0.25, 0.5 , 0.  , 0.25]), array([0.36363636, 0.45454545, 0.18181818]))


In [6]:
# onko tasapainoja enemmän kuin yksi
iterations = 1000
probabilities_A = []
probabilities_B = []

In [7]:
# Simulointi ja lasketaan todennäköisyydet
for i in range(iterations):
    eq = equilibria[0]  # ensimmäinen tasapaino
    probabilities_A.append(eq[0])  # A:n strategian todennäköisyys
    probabilities_B.append(eq[1])  # B:n strategian todennäköisyys

In [8]:
# Simulointi ja lasketaan todennäköisyydet
for i in range(iterations):
    eq = equilibria[0]  # ensimmäinen tasapaino
    probabilities_A.append(eq[0])  # A:n strategian todennäköisyys
    probabilities_B.append(eq[1])  # B:n strategian todennäköisyys

# Sanallinen yhteenveto

Tehtävässä tarkasteltiin peliteorian soveltamista markkinakilpailussa kahden yrityksen, A:n ja B:n, välillä. 

Pelissä on kyse nollasummapelistä, mikä tarkoittaa, että A:n voitot ovat B:n tappioita ja päinvastoin. 

Pelissä on neljä strategiaa A:lle (S1, S2, S3, S4) ja kolme strategiaa B:lle (C1, C2, C3). 

Tässä pelissä A:n strategiat on esitetty markkinaosuuksien muutoksina viikoittain, ja B:n hyödyt ovat käänteisiä A:n arvoille.

Nashin tasapaino

Nashin tasapainot laskettiin NashPy-kirjaston avulla. Lasketut tasapainot osoittavat seuraavat strategioiden todennäköisyydet:

A:n strategiat:

S1 valitaan 25 % todennäköisyydellä

S2 valitaan 50 % todennäköisyydellä

S3 ei valita lainkaan (0 % todennäköisyydellä)

S4 valitaan 25 % todennäköisyydellä

B:n strategiat:

C1 valitaan 36.36 % todennäköisyydellä

C2 valitaan 45.45 % todennäköisyydellä

C3 valitaan 18.18 % todennäköisyydellä

Nashin tasapainossa kummankaan pelaajan ei kannata muuttaa omaa strategiaansa, koska heidän strategiansa tuottavat parhaan mahdollisen hyödyn ottaen huomioon toisen pelaajan strategian. 

Tämä on peliteorian keskeinen periaate: mikään pelaaja ei voi parantaa omaa asemaansa yksipuolisesti, mikä tarkoittaa, että he ovat saavuttaneet tasapainon.

Laskettujen todennäköisyyksien merkitys

A:n todennäköisyydet kertovat, kuinka todennäköisesti A valitsee eri strategioita. 

A valitsee S2 (50 % todennäköisyydellä), S1 ja S4 (molemmat 25 % todennäköisyydellä). 

S3 ei ole valittu, koska se ei ole optimaalinen vaihtoehto.

B:n todennäköisyydet puolestaan osoittavat, kuinka todennäköisesti B valitsee eri strategioita. 

B valitsee C2 (45.45 % todennäköisyydellä), C1 (36.36 %), ja C3 (18.18 %).

# Johtopäätös

Nashin tasapainossa A ja B ovat molemmat optimoinneet strategiansa niin, että kummankaan pelaajan ei ole kannattavaa muuttaa omaa strategiaansa ottaen huomioon toisen pelaajan strategian. 

Tämä tilanne kuvaa strategista tasapainoa, jossa molemmat pelaajat tekevät parhaan mahdollisen päätöksen vastustajan strategian perusteella. 

Peliteoria tarjoaa näin hyödyllisen työkalun ennustamaan kilpailutilanteiden lopputuloksia ja tukemaan strategisten päätösten tekemistä kilpailu- ja markkinatilanteissa.

# Simulointi ja todennäköisyyksien kehitys

Simuloinnilla havaittiin, että todennäköisyydet A:n ja B:n strategioille pysyvät vakiona Nashin tasapainossa. 

Tämä osoittaa, että tasapaino on vakiintunut eikä strategioiden jakauma muutu toistojen myötä. 

Pelissä ei ole tasapainosta poikkeavia strategioita, joten pelaajien strategiat pysyvät ennallaan, kunnes jokin ulkoinen tekijä muuttaa pelin olosuhteita.